py3status/modules/usbguard.py

```diff
@@ -200,182 +82,94 @@ def post_config_hook(self):
-        self.thread.daemon = True
-        # self.thread.start()
 
-    def _start_loop(self):
-        pass
+        thread = Thread(target=lambda: GLib.MainLoop().run())
+        thread.daemon = True
+        thread.start()
 
-    def _get_devices(self, device_filter):
-        if device_filter in ["any", "None"]:
-            device_filter = "match"
-
-        devices = self.proxy.listDevices("(s)", device_filter)
-        new_devices = []
-
-        for device_id, device_string in devices:
-            try:
-                device = self.cache_devices[device_string]
-            except KeyError:
-                device = {"id": device_id}
-                for name, regex in self.device_keys:
-                    value = regex.findall(device_string) or None
-                    if value:
-                        value = (
-                            value[0]
-                            .encode("latin-1")
-                            .decode("unicode_escape")
-                            .encode("latin-1")
-                            .decode()
-                        )
-                    device[name] = value
-                self.cache_devices[device_string] = device
-            new_devices.append(device)
-
-        return deepcopy(new_devices)
+    def _get_devices(self):
+        try:
+            raw_devices = self.proxy.listDevices("(s)", "match")
+        except Exception:
+            raise Exception(STRING_USBGUARD_DBUS)
 
-    def _manipulate_devices(self, devices, device_filter):
-        new_device = []
-        format_device_list = None
-        self.device_length = len(devices)
-        for index, device in enumerate(devices, 1):
-            if device_filter == "any":
-                pass
-            elif device_filter == "None":
-                device["policy"] = None
-            elif device["policy"] != device_filter:
-                continue
-            for x in self.init["format_button"]:
-                composite = self.py3.safe_format(getattr(self, x), device)
-                device[x] = self.py3.composite_update(
-                    composite, {"index": "{}/{}".format(device["id"], x.split("_")[-1])}
+        devices = []
+        for device_id, string in raw_devices:
+            device = {"id": device_id}
+            for name, regex in self.keys:
+                value = regex.findall(string) or None
+                if value:
+                    value = value[0]
+                    value = value.encode("latin-1").decode("unicode_escape")
+                    value = value.encode("latin-1").decode("utf-8")
+                device[name] = value
+            if device["policy"] == "block":
+                devices.append(device)
+
+        return devices
+
+    def _format_device(self, devices):
+        device_info = []
+        for device in devices:
+            for btn in self.init["format_button"]:
+                composite = self.py3.safe_format(getattr(self, btn), device)
```

Comment:
I think you're killing off `format_button_block`. Is it not a useful button for users? You could do `(getattr(self, btn, '{{{}}}'.format{btn), device)` because I think users using `{format_button_block}` would be broken at this part. The code changes it to printing `{button_button_block}`. 

Is it better? I dunno. Users would have to change their config either way. Module breakage vs ugly. Just saying.

Reply:
It is intentional decision to keep only two buttons, "approve" and "reject". As I explained above, "blocked" devices are assumed to be our "TODO items" that need an approval, so a "block" button would literally be a noop.